src/control/cmd/daos/container.go

```diff
@@ -0,0 +1,834 @@
+	DeleteACL    containerDeleteACLCmd    `command:"delete-acl" description:"delete a container's ACL"`
+	SetOwner     containerSetOwnerCmd     `command:"set-owner" description:"change ownership for a container"`
+
+	CreateSnapshot   containerSnapshotCreateCmd   `command:"create-snapshot" alias:"create-snap" description:"create container snapshot"`
+	DestroySnapshot  containerSnapshotDestroyCmd  `command:"destroy-snapshot" alias:"destroy-snap" description:"destroy container snapshot"`
+	ListSnapshots    containerSnapshotListCmd     `command:"list-snapshots" alias:"list-snaps" description:"list container snapshots"`
+	RollbackSnapshot containerSnapshotRollbackCmd `command:"rollback" alias:"rb" description:"roll back container to specified snapshot"`
+}
+
+type containerBaseCmd struct {
+	poolBaseCmd
+	contUUID uuid.UUID
+
+	cContHandle C.daos_handle_t
+}
+
+func (cmd *containerBaseCmd) contUUIDPtr() *C.uchar {
+	return (*C.uchar)(unsafe.Pointer(&cmd.contUUID[0]))
+}
+
+func (cmd *containerBaseCmd) resolveContainer(id ContainerID) error {
+	// TODO: Resolve label.
+	if id.HasLabel() {
+		return errors.New("no support for container labels yet")
+	}
+
+	if !id.HasUUID() {
+		return errors.New("no container UUID provided")
+	}
+	cmd.contUUID = id.UUID
+
+	return nil
+}
+
+func (cmd *containerBaseCmd) openContainer() error {
+	var ci C.daos_cont_info_t
+	rc := C.daos_cont_open(cmd.cPoolHandle, cmd.contUUIDPtr(),
+		C.DAOS_COO_RW|C.DAOS_COO_FORCE, &cmd.cContHandle, &ci, nil)
+	return daosError(rc)
+}
+
+func (cmd *containerBaseCmd) closeContainer() error {
+	cmd.log.Debugf("closing container %s", cmd.contUUID)
+	return daosError(C.daos_cont_close(cmd.cContHandle, nil))
+}
+
+func (cmd *containerBaseCmd) queryContainer() (*containerInfo, error) {
+	ci := newContainerInfo(&cmd.poolUUID, &cmd.contUUID)
+
+	rc := C.daos_cont_query(cmd.cContHandle, &ci.dci, nil, nil)
+	if err := daosError(rc); err != nil {
+		return nil, err
+	}
+
+	return ci, nil
+}
+
+type containerCreateCmd struct {
+	containerBaseCmd
+
+	UUID        string         `long:"cont" short:"c" description:"container UUID (optional)"`
+	Type        string         `long:"type" short:"t" description:"container type" choice:"POSIX" choice:"HDF5" default:"POSIX"`
+	Path        string         `long:"path" short:"d" description:"container namespace path"`
+	ChunkSize   chunkSize      `long:"chunk-size" short:"z" description:"container chunk size"`
+	ObjectClass objectClass    `long:"oclass" short:"o" description:"default object class"`
+	Properties  PropertiesFlag `long:"properties" description:"container properties"`
+	ACLFile     string         `long:"acl-file" short:"A" description:"input file containing ACL"`
+	User        string         `long:"user" short:"u" description:"user who will own the container (username@[domain])"`
+	Group       string         `long:"group" short:"g" description:"group who will own the container (group@[domain])"`
+}
+
```

Comment:
That's the hander for cont create, right?

Reply:
yep